addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -197,18 +197,20 @@ def _send_payment_request(self):
                 "Sending '/payments/create/recurring' request for transaction with reference %s:\n%s",
                 self.reference, pprint.pformat(payload)
             )
+            # Commit transction so it can directly handled by webhook and not create any concurency
+            self.env.cr.commit()
             recurring_payment_data = self.provider_id._razorpay_make_request(
                 'payments/create/recurring', payload=payload
             )
             _logger.info(
                 "Response of '/payments/create/recurring' request for transaction with reference "
                 "%s:\n%s", self.reference, pprint.pformat(recurring_payment_data)
             )
-            self._handle_notification_data('razorpay', recurring_payment_data)
         except ValidationError as e:
             if self.operation == 'offline':
                 self._set_error(str(e))
             else:
+                self.env.cr.rollback()
```

Comment:
Never ever rollback the cursor after sending the payment request. It's the recipe for accidental double payment because the first one left no trace in Odoo and still went through on Razorpay's side.

Reply:
Got it :100: